### ISSUE TITLE
Installation issue: py-tables with Xcode 15

### ISSUE THREAD

**Issue by aphecetche** [LLNL/spack#40223]
### Steps to reproduce the issue

```console
$ spack spec -I <spec>
Input spec
--------------------------------
 -   py-tables
 -       ^mpich

Concretized
--------------------------------
 -   2vf3thv  py-tables@3.7.0%apple-clang@15.0.0~bzip2~lzo+zlib build_system=python_pip arch=darwin-sonoma-m1
[+]  neu3bgj      ^c-blosc@1.21.4%apple-clang@15.0.0+avx2~ipo build_system=cmake build_type=Release generator=make arch=darwin-sonoma-m1
[+]  wvjnpbn          ^cmake@3.27.4%apple-clang@15.0.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=darwin-sonoma-m1
[+]  vahiuh4              ^curl@8.1.2%apple-clang@15.0.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=secure_transport arch=darwin-sonoma-m1
[+]  ooeiwfd                  ^nghttp2@1.52.0%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]  4mmkbnk          ^gmake@4.4.1%apple-clang@15.0.0~guile build_system=autotools arch=darwin-sonoma-m1
[+]  ups4f34          ^lz4@1.9.4%apple-clang@15.0.0+pic build_system=makefile libs=shared,static arch=darwin-sonoma-m1
[+]  cqoamq3          ^snappy@1.1.10%apple-clang@15.0.0~ipo+pic+shared build_system=cmake build_type=Release generator=make arch=darwin-sonoma-m1
[+]  debsblx          ^zstd@1.5.5%apple-clang@15.0.0+programs build_system=makefile compression=none libs=shared,static arch=darwin-sonoma-m1
[+]  uazipiw      ^hdf5@1.14.2%apple-clang@15.0.0~cxx~fortran~hl~ipo~java~map+mpi+shared~szip~threadsafe+tools api=default build_system=cmake build_type=Release generator=make arch=darwin-sonoma-m1
[+]  a6o3jiz          ^mpich@4.1.2%apple-clang@15.0.0~argobots~cuda+fortran+hwloc+hydra+libxml2~pci~rocm+romio~slurm~vci~verbs+wrapperrpath build_system=autotools datatype-engine=auto device=ch4 netmod=ofi pmi=pmi arch=darwin-sonoma-m1
[+]  iqffyhp              ^findutils@4.9.0%apple-clang@15.0.0 build_system=autotools patches=440b954 arch=darwin-sonoma-m1
[+]  h5jblzn              ^hwloc@2.9.1%apple-clang@15.0.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl~pci~rocm build_system=autotools libs=shared,static arch=darwin-sonoma-m1
[+]  t2ehdxh              ^libfabric@1.19.0%apple-clang@15.0.0~debug~kdreg build_system=autotools fabrics=sockets,tcp,udp arch=darwin-sonoma-m1
[+]  g6s6pzk              ^libxml2@2.10.3%apple-clang@15.0.0+pic~python+shared build_system=autotools arch=darwin-sonoma-m1
[+]  3vyj7qy              ^yaksa@0.2%apple-clang@15.0.0~cuda~rocm build_system=autotools arch=darwin-sonoma-m1
[+]  pg324b2                  ^autoconf@2.69%apple-clang@15.0.0 build_system=autotools patches=35c4492,7793209,a49dd5b arch=darwin-sonoma-m1
[+]  egpkhll                  ^automake@1.16.5%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]  ka62m5t                  ^libtool@2.4.7%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]  3kfu6ex                  ^m4@1.4.19%apple-clang@15.0.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=darwin-sonoma-m1
[+]  nstfkxw                      ^libsigsegv@2.14%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]  6n5jqzn          ^pkgconf@1.9.5%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]  6ivghe6      ^py-cython@0.29.36%apple-clang@15.0.0 build_system=python_pip patches=c4369ad arch=darwin-sonoma-m1
[+]  tcm4ils      ^py-numexpr@2.8.4%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]  nw2rm2m      ^py-numpy@1.25.2%apple-clang@15.0.0+blas+lapack build_system=python_pip patches=873745d arch=darwin-sonoma-m1
[+]  vz27cva          ^openblas@0.3.24%apple-clang@15.0.0~bignuma~consistent_fpcsr+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=none arch=darwin-sonoma-m1
[+]  42owqpl              ^perl@5.38.0%apple-clang@15.0.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=darwin-sonoma-m1
[+]  5p4t5n2                  ^berkeley-db@18.1.40%apple-clang@15.0.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=darwin-sonoma-m1
[+]  364jwea      ^py-packaging@23.1%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]  hrzfnul          ^py-flit-core@3.9.0%apple-clang@15.0.0 build_system=python_pip arch=darwin-sonoma-m1
[+]  52663z5      ^py-pip@23.1.2%apple-clang@15.0.0 build_system=generic arch=darwin-sonoma-m1
[+]  kh6fu55      ^py-setuptools@63.4.3%apple-clang@15.0.0 build_system=generic arch=darwin-sonoma-m1
[+]  34rqb5g      ^py-wheel@0.37.1%apple-clang@15.0.0 build_system=generic arch=darwin-sonoma-m1
[+]  lhp4g2v      ^python@3.11.4%apple-clang@15.0.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060 arch=darwin-sonoma-m1
[e]  mzknptf          ^apple-libuuid@1353.100.2%apple-clang@15.0.0 build_system=bundle arch=darwin-sonoma-m1
[+]  56nm4nu          ^bzip2@1.0.8%apple-clang@15.0.0~debug~pic+shared build_system=generic arch=darwin-sonoma-m1
[+]  g525eux              ^diffutils@3.9%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]  h3fuhrh          ^expat@2.5.0%apple-clang@15.0.0~libbsd build_system=autotools arch=darwin-sonoma-m1
[+]  aiv7vig          ^gdbm@1.23%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]  64y7b63          ^gettext@0.21.1%apple-clang@15.0.0+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=darwin-sonoma-m1
[+]  pgvu7yw              ^libiconv@1.17%apple-clang@15.0.0 build_system=autotools libs=shared,static arch=darwin-sonoma-m1
[+]  ge5hjwc              ^tar@1.34%apple-clang@15.0.0 build_system=autotools zip=pigz arch=darwin-sonoma-m1
[+]  cjsrdl6                  ^pigz@2.7%apple-clang@15.0.0 build_system=makefile arch=darwin-sonoma-m1
[+]  bjxi4tt          ^libffi@3.4.4%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m1
[+]  6dtvhnr          ^libxcrypt@4.4.35%apple-clang@15.0.0~obsolete_api build_system=autotools patches=4885da3 arch=darwin-sonoma-m1
[+]  yqwnbtb          ^ncurses@6.4%apple-clang@15.0.0~symlinks+termlib abi=none build_system=autotools arch=darwin-sonoma-m1
[+]  gcaqi63          ^openssl@3.1.3%apple-clang@15.0.0~docs+shared build_system=generic certs=mozilla arch=darwin-sonoma-m1
[+]  sbr7xj6              ^ca-certificates-mozilla@2023-05-30%apple-clang@15.0.0 build_system=generic arch=darwin-sonoma-m1
[+]  zooxgts          ^readline@8.2%apple-clang@15.0.0 build_system=autotools patches=bbf97f1 arch=darwin-sonoma-m1
[+]  vrq62li          ^sqlite@3.42.0%apple-clang@15.0.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=darwin-sonoma-m1
[+]  fagghrq          ^xz@5.4.1%apple-clang@15.0.0~pic build_system=autotools libs=shared,static arch=darwin-sonoma-m1
[+]  6dno2lc      ^zlib-ng@2.1.3%apple-clang@15.0.0+compat+opt build_system=autotools patches=299b958,86c3f5b,ae9077a,b692621 arch=darwin-sonoma-m1
[+]  7gytrfu          ^gnuconfig@2022-09-17%apple-clang@15.0.0 build_system=generic arch=darwin-sonoma-m1

```

### Error message

<details><summary>Error message</summary>

<pre>
 /var/folders/wt/gcp_6d6122s8yw07_6wzhw6w0000gp/T/lzo_version_datefn9yoc8n.c:
            2:5: error: call to undeclared function 'lzo_version_date'; ISO C99 and late
            r do not support implicit function declarations [-Wimplicit-function-declara
            tion]
     18           lzo_version_date();

</pre></details>

### Information on your system

* **Spack:** 0.21.0.dev0 (5165524ca6eb9cb0b8c2e225352e4729b5a04062)
* **Python:** 3.9.6
* **Platform:** darwin-sonoma-m1
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/12739723/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/12739724/spack-build-out.txt)

@adamjstewart 

(note that I'm trying with ^mpich as otherwise the build is failing within openmpi. In any case I'm not interested in any MPI implementation on my Mac, just want py-tables to compile :wink: ) 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by adamjstewart** [comment]
Investigating now.

> I'm not interested in any MPI implementation on my Mac, just want py-tables to compile

You could use `py-tables ^hdf5~mpi` or even disable MPI globally in `packages.yaml`:
```yaml
packages:
  all:
    variants: ~mpi
```

**Comment by adamjstewart** [comment]
I tried building the latest version of the package but that has the same issue. Can you open an issue upstream and see if they know how to fix this issue?

**Comment by aphecetche** [comment]
Forgot to say that I've tried the master (cfa1e911) from sep 26 and got it built.

**Comment by adamjstewart** [comment]
This was fixed in py-tables 3.9.0 which was added in #40340
